gtsfm/evaluation/metrics_report.py

```diff
@@ -179,25 +253,43 @@ def get_html_header() -> str:
               </head>"""
 
 
-def generate_metrics_report_html(metrics_groups: List[GtsfmMetricsGroup], html_path: str) -> None:
+def generate_metrics_report_html(
+    metrics_groups: List[GtsfmMetricsGroup],
+    html_path: str,
+    colmap_output_dir: str,
+    metric_paths: List[str],
+) -> None:
     """Generates a report for metrics groups with plots and tables and saves it to HTML.
 
     Args:
         metrics_groups: List of metrics to be reported.
+        html_path: Path where this report is written to (default: output_dir/gtsfm_metrics_report.html).
+        colmap_output_dir: Optional; If a path to a directory containing a COLMAP reconstruction
+          (as cameras.txt, images.txt, and points3D.txt) is provided, the COLMAP metrics will also be
+          included in the report.
+        metric_paths: A list of paths to GTSfM metrics.
```

Comment:
the function seems to be doing too many things. It is supposed to generate a HTML from known metrics. Instead of passing in the colmap path to it, I recommend passing in colmap GtsfmMetricGroups. 
The computation of these metric groups can be done in another function (which is not called by this).

Reply:
If youre curious, here are some of the issues I see with this design: 
- it is asking for the GTSfM metric paths although it already has GTSfM metrics. 
- it is assuming the colmap metrics are situated are in a specific location with respect to gtsfm metrics. 
- its job is to generate a html, not to process colmap output :)

Reply:
This function now receives colmap GtsfmMetricGroups, which are computed in another function.